controller.py

```diff
@@ -129,11 +141,29 @@ def update_url(time=last_update_time, page=None):
 
         return feed_response(update_feed)
 
-    def remove_items(self):
-        catalog = self.authenticated_catalog_from_request()
-        if isinstance(catalog, ProblemDetail):
-            return catalog
+    def register_client(self):
```

Comment:
I'm not comfortable registering clients through an open API call. In the long run I think the simplest thing would be a callback verification process. If you try to register http://foo/ with the metadata wrangler, the metadata wrangler should POST the secret to http://foo/metadata_wrangler_callback (or some similar URL), and only save the registration if the callback succeeds. 

I'm not saying you should implement this now--it's not necessary given the small number of libraries we're starting with, and it might have security implications I haven't considered.

Unless you do want to implement something like that now (in which case we can talk about an implementation), I would take this bit out for now. I think it's simpler to add clients to the metadata wrangler by running the script and communicating the key to the library out-of-band.